src/builder.rs

```diff
@@ -112,12 +88,12 @@ pub fn run(root_nix_file: &NixFile, cas: &ContentAddressable) -> Result<Info, Er
 enum LogDatum {
     Source(PathBuf),
     AttrDrv(String, PathBuf),
-    Text(String),
+    Text(OsString),
 }
 
 /// Examine a line of output and extract interesting log items in to
 /// structured data.
-fn parse_evaluation_line(line: &str) -> LogDatum {
+fn parse_evaluation_line(line: &OsStr) -> LogDatum {
```

Comment:
Instead of making this take an `&OsStr` let's make it take an `AsRef<OsStr>`, since it'll make the tests much nicer.

Here I've written a playground example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ba3c83196bc09d155e2fe40b77447c73

The key change is the signature:

```
fn parse_evaluation_line<T>(line: T) -> LogDatum
where
    T: AsRef<OsStr>,
{
```

and then:

```
    let line: &OsStr = line.as_ref();
```

after that, none of the tests should need to change.